src/lib/operators/abstract_join_operator.cpp

```diff
@@ -8,8 +8,8 @@
 
 namespace opossum {
 
-AbstractJoinOperator::AbstractJoinOperator(const std::shared_ptr<const AbstractOperator> left,
-                                           const std::shared_ptr<const AbstractOperator> right, const JoinMode mode,
+AbstractJoinOperator::AbstractJoinOperator(const std::shared_ptr<const AbstractOperator>& left,
+                                           const std::shared_ptr<const AbstractOperator>& right, const JoinMode mode,
```

Comment:
Disagreed - each operator has to have shared ownership of the previous operators.